frontend_server/lib/server.dart

```diff
@@ -137,7 +136,7 @@ class _FrontendCompiler implements CompilerInterface {
     ArgResults options, {
     IncrementalKernelGenerator generator,
   }) async {
-    _filename = filename;
+    Uri _filenameUri = Uri.base.resolve(new Uri.file(filename).toString());
```

Comment:
Consider using Uri.base.resolveUri(new Uri.file(filename)) to avoid converting URI to String and parsing it back to URI.